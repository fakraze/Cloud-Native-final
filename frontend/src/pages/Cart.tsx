import React from 'react';
import { Link, useNavigate } from 'react-router-dom';
import { ShoppingCart, Plus, Minus, Trash2, ArrowLeft } from 'lucide-react';
import { useCartStore } from '../store/cartStore';
import { useUpdateCartItem, useRemoveFromCart, useCreateOrder } from '../hooks/useOrder';

const Cart: React.FC = () => {
  const { cart } = useCartStore();
  const navigate = useNavigate();
  const updateCartMutation = useUpdateCartItem();
  const removeFromCartMutation = useRemoveFromCart();
  const createOrderMutation = useCreateOrder();

  if (!cart || cart.items.length === 0) {
    return (
      <div className="max-w-4xl mx-auto">
        <div className="text-center py-16">
          <ShoppingCart className="h-16 w-16 text-gray-300 mx-auto mb-4" />
          <h2 className="text-2xl font-semibold text-gray-900 mb-2">Your cart is empty</h2>
          <p className="text-gray-600 mb-6">Add some delicious items to get started!</p>
          <Link to="/restaurant" className="btn-primary">
            Browse Restaurants
          </Link>
        </div>
      </div>
    );
  }

  const updateQuantity = (itemId: string, newQuantity: number) => {
    if (newQuantity === 0) {
      removeFromCartMutation.mutate(itemId);
    } else {
      updateCartMutation.mutate({
        cartItemId: itemId,
        updates: { quantity: newQuantity }
      });
    }
  };

  const removeItem = (itemId: string) => {
    removeFromCartMutation.mutate(itemId);
  };
  const handleCheckout = () => {
    console.log(cart);
    const orderData = {      
      restaurantId: cart.items[0].menuItem.restaurantId,
      userId: cart.userId,
      items: cart.items.map(item => ({
        menuItemId: item.menuItem.id,
        menuItemName: item.menuItem.name,
        quantity: item.quantity,
        price: item.menuItem.price,
        customizations: item.customizations,
        notes: item.notes
      })),
      totalAmount: cart.totalAmount,
      deliveryType: 'pickup' as const
    };

    createOrderMutation.mutate(orderData, {
      onSuccess: (order) => {
        navigate(`/order/${order.id}`);
      }
    });
  };

  return (
    <div className="max-w-4xl mx-auto">
      {/* Header */}
      <div className="flex items-center justify-between mb-8">
        <div className="flex items-center space-x-4">
          <Link 
            to="/restaurant" 
            className="inline-flex items-center space-x-2 text-gray-600 hover:text-gray-900"
          >
            <ArrowLeft className="h-5 w-5" />
            <span>Continue Shopping</span>
          </Link>
        </div>
        <h1 className="text-3xl font-bold text-gray-900">Shopping Cart</h1>
      </div>

      <div className="grid grid-cols-1 lg:grid-cols-3 gap-8">
        {/* Cart Items */}
        <div className="lg:col-span-2 space-y-4">
          {cart.items.map((item) => (
            <div key={item.id} className="card">
              <div className="flex items-center space-x-4">
                <img
                  src={item.menuItem.imageUrl || '/placeholder-food.jpg'}
                  alt={item.menuItem.name}
                  className="w-20 h-20 object-cover rounded-lg"
                />
                
                <div className="flex-1">
                  <h3 className="font-semibold text-gray-900">{item.menuItem.name}</h3>
                  <p className="text-sm text-gray-600 mb-2">{item.menuItem.description}</p>
                  
                  {/* Customizations */}
                  {Object.keys(item.customizations).length > 0 && (
                    <div className="text-xs text-gray-500 mb-2">
                      {Object.entries(item.customizations).map(([key, value]) => (
                        <div key={key}>
                          {Array.isArray(value) ? value.join(', ') : value}
                        </div>
                      ))}
                    </div>
                  )}
                  
                  {item.notes && (
                    <p className="text-xs text-gray-500 italic">Note: {item.notes}</p>
                  )}
                </div>
                
                <div className="text-right">
                  <div className="text-lg font-semibold text-gray-900 mb-2">
                    ${(item.menuItem.price * item.quantity).toFixed(2)}
                  </div>
                  
                  <div className="flex items-center space-x-2">
                    <button
                      onClick={() => updateQuantity(item.id, item.quantity - 1)}
                      className="p-1 rounded-full border border-gray-300 hover:bg-gray-50"
                      disabled={updateCartMutation.isPending}
                    >
                      <Minus className="h-4 w-4" />
                    </button>
                    <span className="w-8 text-center font-medium">{item.quantity}</span>
                    <button
                      onClick={() => updateQuantity(item.id, item.quantity + 1)}
                      className="p-1 rounded-full border border-gray-300 hover:bg-gray-50"
                      disabled={updateCartMutation.isPending}
                    >
                      <Plus className="h-4 w-4" />
                    </button>
                    <button
                      onClick={() => removeItem(item.id)}
                      className="p-1 text-red-500 hover:text-red-700"
                      disabled={removeFromCartMutation.isPending}
                    >
                      <Trash2 className="h-4 w-4" />
                    </button>
                  </div>
                </div>
              </div>
            </div>
          ))}
        </div>

        {/* Order Summary */}
        <div className="lg:col-span-1">
          <div className="card sticky top-4">
            <h3 className="text-lg font-semibold text-gray-900 mb-4">Order Summary</h3>
            
            <div className="space-y-2 mb-4">
              <div className="flex justify-between text-sm">
                <span>Subtotal</span>
                <span>${cart.totalAmount.toFixed(2)}</span>
              </div>
              <div className="flex justify-between text-sm">
                <span>Tax (8%)</span>
                <span>${(cart.totalAmount * 0.08).toFixed(2)}</span>
              </div>
              <div className="border-t pt-2 flex justify-between font-semibold">
                <span>Total</span>
                <span>${(cart.totalAmount * 1.08).toFixed(2)}</span>
              </div>
            </div>
            
            <button
              onClick={handleCheckout}
              disabled={createOrderMutation.isPending}
              className={`w-full btn-primary ${
                createOrderMutation.isPending ? 'opacity-50 cursor-not-allowed' : ''
              }`}
            >
              {createOrderMutation.isPending ? 'Processing...' : 'Checkout'}
            </button>
            
            {createOrderMutation.error && (
              <div className="mt-3 bg-red-50 border border-red-200 rounded p-2">
                <p className="text-sm text-red-600">Failed to create order. Please try again.</p>
              </div>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export { Cart };
