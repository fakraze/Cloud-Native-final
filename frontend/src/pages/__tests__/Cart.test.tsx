import { describe, it, expect, beforeEach, vi } from 'vitest';
import { render, screen, waitFor } from '../../test/test-utils';
import userEvent from '@testing-library/user-event';
import { Cart } from '../Cart';
import { useNavigate } from 'react-router-dom';

// Mock react-router-dom
vi.mock('react-router-dom', async () => {
  const actual = await vi.importActual('react-router-dom');
  return {
    ...actual,
    useNavigate: vi.fn(),
  };
});

// Mock the hooks and stores
const mockCartStore = vi.hoisted(() => ({
  useCartStore: vi.fn()
}));

vi.mock('../../store/cartStore', () => mockCartStore);

describe('Cart Component', () => {
  const mockNavigate = vi.fn();
  const mockUpdateCart = vi.fn();
  const mockRemoveFromCart = vi.fn();
  const mockCreateOrder = vi.fn();

  beforeEach(() => {
    vi.clearAllMocks();
    vi.mocked(useNavigate).mockReturnValue(mockNavigate);
    
    vi.mocked(require('../../hooks/useOrder').useUpdateCartItem).mockReturnValue({
      mutate: mockUpdateCart
    });
    
    vi.mocked(require('../../hooks/useOrder').useRemoveFromCart).mockReturnValue({
      mutate: mockRemoveFromCart
    });
    
    vi.mocked(require('../../hooks/useOrder').useCreateOrder).mockReturnValue({
      mutate: mockCreateOrder,
      isLoading: false
    });
  });

  describe('Empty Cart State', () => {
    it('should render empty cart when cart is null', () => {
      mockCartStore.useCartStore.mockReturnValue({
        cart: null
      });

      render(<Cart />);
      
      expect(screen.getByText('Your cart is empty')).toBeInTheDocument();
      expect(screen.getByText('Add some delicious items to get started!')).toBeInTheDocument();
      expect(screen.getByText('Browse Restaurants')).toBeInTheDocument();
    });

    it('should render empty cart when cart has no items', () => {
      mockCartStore.useCartStore.mockReturnValue({
        cart: { id: '1', items: [], restaurantId: 'rest-1' }
      });

      render(<Cart />);
      
      expect(screen.getByText('Your cart is empty')).toBeInTheDocument();
    });

    it('should navigate to restaurants when browse button is clicked', async () => {
      const user = userEvent.setup();
      mockCartStore.useCartStore.mockReturnValue({
        cart: null
      });

      render(<Cart />);
      
      const browseButton = screen.getByText('Browse Restaurants');
      await user.click(browseButton);
      
      // Since it's a Link, it won't call navigate directly
      expect(browseButton).toHaveAttribute('href', '/restaurant');
    });
  });

  describe('Cart with Items', () => {
    const mockCartWithItems = {
      id: '1',
      restaurantId: 'rest-1',
      items: [
        {
          id: 'item-1',
          quantity: 2,
          customizations: {},
          menuItem: {
            id: 'menu-1',
            name: 'Pizza Margherita',
            price: 12.99,
            imageUrl: 'pizza.jpg',
            restaurantId: 'rest-1'
          }
        },
        {
          id: 'item-2',
          quantity: 1,
          customizations: {},
          menuItem: {
            id: 'menu-2',
            name: 'Caesar Salad',
            price: 8.99,
            imageUrl: 'salad.jpg',
            restaurantId: 'rest-1'
          }
        }
      ]
    };

    beforeEach(() => {
      mockCartStore.useCartStore.mockReturnValue({
        cart: mockCartWithItems
      });
    });

    it('should render cart items correctly', () => {
      render(<Cart />);
      
      expect(screen.getByText('Pizza Margherita')).toBeInTheDocument();
      expect(screen.getByText('Caesar Salad')).toBeInTheDocument();
      expect(screen.getByText('$12.99')).toBeInTheDocument();
      expect(screen.getByText('$8.99')).toBeInTheDocument();
    });

    it('should display correct quantities', () => {
      render(<Cart />);
      
      const quantityInputs = screen.getAllByRole('spinbutton');
      expect(quantityInputs[0]).toHaveValue(2);
      expect(quantityInputs[1]).toHaveValue(1);
    });

    it('should calculate total correctly', () => {
      render(<Cart />);
      
      // Total should be (12.99 * 2) + (8.99 * 1) = 34.97
      expect(screen.getByText('$34.97')).toBeInTheDocument();
    });

    it('should update quantity when plus button is clicked', async () => {
      const user = userEvent.setup();
      render(<Cart />);
      
      const plusButtons = screen.getAllByLabelText(/increase quantity/i);
      await user.click(plusButtons[0]);
      
      expect(mockUpdateCart).toHaveBeenCalledWith({
        cartItemId: 'item-1',
        updates: { quantity: 3 }
      });
    });

    it('should update quantity when minus button is clicked', async () => {
      const user = userEvent.setup();
      render(<Cart />);
      
      const minusButtons = screen.getAllByLabelText(/decrease quantity/i);
      await user.click(minusButtons[0]);
      
      expect(mockUpdateCart).toHaveBeenCalledWith({
        cartItemId: 'item-1',
        updates: { quantity: 1 }
      });
    });

    it('should remove item when quantity becomes 0', async () => {
      const user = userEvent.setup();
      
      // Set up cart with item having quantity 1
      const cartWithOneItem = {
        ...mockCartWithItems,
        items: [{
          ...mockCartWithItems.items[0],
          quantity: 1
        }]
      };
      
      mockCartStore.useCartStore.mockReturnValue({
        cart: cartWithOneItem
      });
      
      render(<Cart />);
      
      const minusButton = screen.getByLabelText(/decrease quantity/i);
      await user.click(minusButton);
      
      expect(mockRemoveFromCart).toHaveBeenCalledWith('item-1');
    });

    it('should remove item when delete button is clicked', async () => {
      const user = userEvent.setup();
      render(<Cart />);
      
      const deleteButtons = screen.getAllByLabelText(/remove item/i);
      await user.click(deleteButtons[0]);
      
      expect(mockRemoveFromCart).toHaveBeenCalledWith('item-1');
    });

    it('should handle checkout process', async () => {
      const user = userEvent.setup();
      render(<Cart />);
      
      const checkoutButton = screen.getByText(/proceed to checkout/i);
      await user.click(checkoutButton);
      
      expect(mockCreateOrder).toHaveBeenCalledWith(
        expect.objectContaining({
          restaurantId: 'rest-1',
          items: expect.arrayContaining([
            expect.objectContaining({
              menuItemId: 'menu-1',
              quantity: 2
            })
          ])
        })
      );
    });

    it('should show loading state during checkout', () => {
      vi.mocked(require('../../hooks/useOrder').useCreateOrder).mockReturnValue({
        mutate: mockCreateOrder,
        isLoading: true
      });

      render(<Cart />);
      
      const checkoutButton = screen.getByRole('button', { name: /processing/i });
      expect(checkoutButton).toBeDisabled();
    });
  });

  describe('Navigation', () => {
    it('should have back to restaurant button', () => {
      mockCartStore.useCartStore.mockReturnValue({
        cart: null
      });

      render(<Cart />);
      
      const backButton = screen.getByLabelText(/back to restaurants/i);
      expect(backButton).toBeInTheDocument();
    });
  });

  describe('Snapshots', () => {
    it('should match snapshot for empty cart', () => {
      mockCartStore.useCartStore.mockReturnValue({
        cart: null
      });

      const { container } = render(<Cart />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for cart with items', () => {
      mockCartStore.useCartStore.mockReturnValue({
        cart: {
          id: '1',
          restaurantId: 'rest-1',
          items: [
            {
              id: 'item-1',
              quantity: 2,
              customizations: {},
              menuItem: {
                id: 'menu-1',
                name: 'Pizza Margherita',
                price: 12.99,
                imageUrl: 'pizza.jpg',
                restaurantId: 'rest-1'
              }
            }
          ]
        }
      });

      const { container } = render(<Cart />);
      expect(container.firstChild).toMatchSnapshot();
    });

    it('should match snapshot for loading state', () => {
      vi.mocked(require('../../hooks/useOrder').useCreateOrder).mockReturnValue({
        mutate: mockCreateOrder,
        isLoading: true
      });

      mockCartStore.useCartStore.mockReturnValue({
        cart: {
          id: '1',
          restaurantId: 'rest-1',
          items: [
            {
              id: 'item-1',
              quantity: 1,
              customizations: {},
              menuItem: {
                id: 'menu-1',
                name: 'Test Item',
                price: 10.99,
                imageUrl: 'test.jpg',
                restaurantId: 'rest-1'
              }
            }
          ]
        }
      });

      const { container } = render(<Cart />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
              imageUrl: '/images/pizza.jpg'
            }
          },
          {
            id: '2',
            name: 'Caesar Salad',
            price: 8.50,
            quantity: 1,
            restaurantId: 'rest-1',
            restaurantName: 'Italiano Restaurant',
            customizations: { 'dressing': 'Caesar', 'croutons': 'Extra' },
            menuItem: {
              id: '2',
              name: 'Caesar Salad',
              imageUrl: '/images/salad.jpg'
            }
          }
        ],
        total: 34.48,
        totalAmount: 34.48
      }
    });

    const { container } = render(<Cart />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render cart with single item', () => {
    mockCartStore.useCartStore.mockReturnValue({
      cart: {
        id: '1',
        items: [
          {
            id: '1',
            name: 'Burger Deluxe',
            price: 15.99,
            quantity: 1,
            restaurantId: 'rest-2',
            restaurantName: 'Burger House',
            customizations: {},
            menuItem: {
              id: '1',
              name: 'Burger Deluxe',
              imageUrl: '/images/burger.jpg'
            }
          }
        ],
        total: 15.99,
        totalAmount: 15.99
      }
    });

    const { container } = render(<Cart />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
