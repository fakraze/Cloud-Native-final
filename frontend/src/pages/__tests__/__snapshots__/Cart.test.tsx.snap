// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart Component > should render cart with items 1`] = `
<div
  class="max-w-4xl mx-auto"
>
  <div
    class="flex items-center justify-between mb-8"
  >
    <div
      class="flex items-center space-x-4"
    >
      <a
        class="inline-flex items-center space-x-2 text-gray-600 hover:text-gray-900"
        data-discover="true"
        href="/restaurant"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-left h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m12 19-7-7 7-7"
          />
          <path
            d="M19 12H5"
          />
        </svg>
        <span>
          Continue Shopping
        </span>
      </a>
    </div>
    <h1
      class="text-3xl font-bold text-gray-900"
    >
      Shopping Cart
    </h1>
  </div>
  <div
    class="grid grid-cols-1 lg:grid-cols-3 gap-8"
  >
    <div
      class="lg:col-span-2 space-y-4"
    >
      <div
        class="card"
      >
        <div
          class="flex items-center space-x-4"
        >
          <img
            alt="Pizza Margherita"
            class="w-20 h-20 object-cover rounded-lg"
            src="/images/pizza.jpg"
          />
          <div
            class="flex-1"
          >
            <h3
              class="font-semibold text-gray-900"
            >
              Pizza Margherita
            </h3>
            <p
              class="text-sm text-gray-600 mb-2"
            />
          </div>
          <div
            class="text-right"
          >
            <div
              class="text-lg font-semibold text-gray-900 mb-2"
            >
              $
              NaN
            </div>
            <div
              class="flex items-center space-x-2"
            >
              <button
                class="p-1 rounded-full border border-gray-300 hover:bg-gray-50"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-minus h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                </svg>
              </button>
              <span
                class="w-8 text-center font-medium"
              >
                2
              </span>
              <button
                class="p-1 rounded-full border border-gray-300 hover:bg-gray-50"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-plus h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </button>
              <button
                class="p-1 text-red-500 hover:text-red-700"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-trash2 lucide-trash-2 h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 6h18"
                  />
                  <path
                    d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                  />
                  <path
                    d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                  />
                  <line
                    x1="10"
                    x2="10"
                    y1="11"
                    y2="17"
                  />
                  <line
                    x1="14"
                    x2="14"
                    y1="11"
                    y2="17"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="card"
      >
        <div
          class="flex items-center space-x-4"
        >
          <img
            alt="Caesar Salad"
            class="w-20 h-20 object-cover rounded-lg"
            src="/images/salad.jpg"
          />
          <div
            class="flex-1"
          >
            <h3
              class="font-semibold text-gray-900"
            >
              Caesar Salad
            </h3>
            <p
              class="text-sm text-gray-600 mb-2"
            />
            <div
              class="text-xs text-gray-500 mb-2"
            >
              <div>
                Caesar
              </div>
              <div>
                Extra
              </div>
            </div>
          </div>
          <div
            class="text-right"
          >
            <div
              class="text-lg font-semibold text-gray-900 mb-2"
            >
              $
              NaN
            </div>
            <div
              class="flex items-center space-x-2"
            >
              <button
                class="p-1 rounded-full border border-gray-300 hover:bg-gray-50"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-minus h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                </svg>
              </button>
              <span
                class="w-8 text-center font-medium"
              >
                1
              </span>
              <button
                class="p-1 rounded-full border border-gray-300 hover:bg-gray-50"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-plus h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </button>
              <button
                class="p-1 text-red-500 hover:text-red-700"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-trash2 lucide-trash-2 h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 6h18"
                  />
                  <path
                    d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                  />
                  <path
                    d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                  />
                  <line
                    x1="10"
                    x2="10"
                    y1="11"
                    y2="17"
                  />
                  <line
                    x1="14"
                    x2="14"
                    y1="11"
                    y2="17"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="lg:col-span-1"
    >
      <div
        class="card sticky top-4"
      >
        <h3
          class="text-lg font-semibold text-gray-900 mb-4"
        >
          Order Summary
        </h3>
        <div
          class="space-y-2 mb-4"
        >
          <div
            class="flex justify-between text-sm"
          >
            <span>
              Subtotal
            </span>
            <span>
              $
              34.48
            </span>
          </div>
          <div
            class="flex justify-between text-sm"
          >
            <span>
              Tax (8%)
            </span>
            <span>
              $
              2.76
            </span>
          </div>
          <div
            class="border-t pt-2 flex justify-between font-semibold"
          >
            <span>
              Total
            </span>
            <span>
              $
              37.24
            </span>
          </div>
        </div>
        <button
          class="w-full btn-primary "
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cart Component > should render cart with single item 1`] = `
<div
  class="max-w-4xl mx-auto"
>
  <div
    class="flex items-center justify-between mb-8"
  >
    <div
      class="flex items-center space-x-4"
    >
      <a
        class="inline-flex items-center space-x-2 text-gray-600 hover:text-gray-900"
        data-discover="true"
        href="/restaurant"
      >
        <svg
          aria-hidden="true"
          class="lucide lucide-arrow-left h-5 w-5"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m12 19-7-7 7-7"
          />
          <path
            d="M19 12H5"
          />
        </svg>
        <span>
          Continue Shopping
        </span>
      </a>
    </div>
    <h1
      class="text-3xl font-bold text-gray-900"
    >
      Shopping Cart
    </h1>
  </div>
  <div
    class="grid grid-cols-1 lg:grid-cols-3 gap-8"
  >
    <div
      class="lg:col-span-2 space-y-4"
    >
      <div
        class="card"
      >
        <div
          class="flex items-center space-x-4"
        >
          <img
            alt="Burger Deluxe"
            class="w-20 h-20 object-cover rounded-lg"
            src="/images/burger.jpg"
          />
          <div
            class="flex-1"
          >
            <h3
              class="font-semibold text-gray-900"
            >
              Burger Deluxe
            </h3>
            <p
              class="text-sm text-gray-600 mb-2"
            />
          </div>
          <div
            class="text-right"
          >
            <div
              class="text-lg font-semibold text-gray-900 mb-2"
            >
              $
              NaN
            </div>
            <div
              class="flex items-center space-x-2"
            >
              <button
                class="p-1 rounded-full border border-gray-300 hover:bg-gray-50"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-minus h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                </svg>
              </button>
              <span
                class="w-8 text-center font-medium"
              >
                1
              </span>
              <button
                class="p-1 rounded-full border border-gray-300 hover:bg-gray-50"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-plus h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M5 12h14"
                  />
                  <path
                    d="M12 5v14"
                  />
                </svg>
              </button>
              <button
                class="p-1 text-red-500 hover:text-red-700"
              >
                <svg
                  aria-hidden="true"
                  class="lucide lucide-trash2 lucide-trash-2 h-4 w-4"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 6h18"
                  />
                  <path
                    d="M19 6v14c0 1-1 2-2 2H7c-1 0-2-1-2-2V6"
                  />
                  <path
                    d="M8 6V4c0-1 1-2 2-2h4c1 0 2 1 2 2v2"
                  />
                  <line
                    x1="10"
                    x2="10"
                    y1="11"
                    y2="17"
                  />
                  <line
                    x1="14"
                    x2="14"
                    y1="11"
                    y2="17"
                  />
                </svg>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="lg:col-span-1"
    >
      <div
        class="card sticky top-4"
      >
        <h3
          class="text-lg font-semibold text-gray-900 mb-4"
        >
          Order Summary
        </h3>
        <div
          class="space-y-2 mb-4"
        >
          <div
            class="flex justify-between text-sm"
          >
            <span>
              Subtotal
            </span>
            <span>
              $
              15.99
            </span>
          </div>
          <div
            class="flex justify-between text-sm"
          >
            <span>
              Tax (8%)
            </span>
            <span>
              $
              1.28
            </span>
          </div>
          <div
            class="border-t pt-2 flex justify-between font-semibold"
          >
            <span>
              Total
            </span>
            <span>
              $
              17.27
            </span>
          </div>
        </div>
        <button
          class="w-full btn-primary "
        >
          Checkout
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Cart Component > should render empty cart state 1`] = `
<div
  class="max-w-4xl mx-auto"
>
  <div
    class="text-center py-16"
  >
    <svg
      aria-hidden="true"
      class="lucide lucide-shopping-cart h-16 w-16 text-gray-300 mx-auto mb-4"
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <circle
        cx="8"
        cy="21"
        r="1"
      />
      <circle
        cx="19"
        cy="21"
        r="1"
      />
      <path
        d="M2.05 2.05h2l2.66 12.42a2 2 0 0 0 2 1.58h9.78a2 2 0 0 0 1.95-1.57l1.65-7.43H5.12"
      />
    </svg>
    <h2
      class="text-2xl font-semibold text-gray-900 mb-2"
    >
      Your cart is empty
    </h2>
    <p
      class="text-gray-600 mb-6"
    >
      Add some delicious items to get started!
    </p>
    <a
      class="btn-primary"
      data-discover="true"
      href="/restaurant"
    >
      Browse Restaurants
    </a>
  </div>
</div>
`;
